integration_test/playertesting/expectations/MultipleEventsExpectation.ts

```diff
@@ -0,0 +1,175 @@
+      (singleExpectationConvertible) =>
+        singleExpectationConvertible instanceof SingleEventExpectation
+          ? singleExpectationConvertible
+          : new PlainEventExpectation(singleExpectationConvertible as EventType)
+    );
+  }
+
+  get expectedFulfillmentCount(): number {
+    return this.singleExpectations.length;
+  }
+
+  isNextExpectationMet(receivedEvent: Event): boolean {
+    const nextUnfulfilled = this.singleExpectations.find(
+      (expectation) =>
+        !expectation.isFulfilled &&
+        expectation.maybeFulfillExpectation(receivedEvent)
+    );
+
+    return nextUnfulfilled?.isFulfilled ?? false;
+  }
+
+  toString(): string {
+    return this.singleExpectations
+      .map((expectation) => expectation.toString())
+      .join(' - ');
+  }
+}
+
+// To expect a given eventType to happen multiple times.
+export class RepeatedEventExpectation extends EventSequenceExpectation {
+  constructor(
+    singleExpectationConvertible: SingleEventExpectation | EventType,
+    count: number
+  ) {
+    const singleExpectation =
+      singleExpectationConvertible instanceof SingleEventExpectation
+        ? singleExpectationConvertible
+        : new PlainEventExpectation(singleExpectationConvertible as EventType);
+    super(Array.from({ length: count }, () => singleExpectation.copy()));
+  }
+}
+
+// To expect at least one out of multiple events.
+export class AnyEventExpectation extends MultipleEventsExpectation {
+  readonly singleExpectations: SingleEventExpectation[];
+
+  constructor(
+    singleExpectationConvertibles: SingleEventExpectation[] | EventType[]
+  ) {
+    super();
+    this.singleExpectations = singleExpectationConvertibles.map(
+      (singleExpectationConvertible) =>
+        singleExpectationConvertible instanceof SingleEventExpectation
+          ? singleExpectationConvertible
+          : new PlainEventExpectation(singleExpectationConvertible as EventType)
+    );
+  }
+
+  get expectedFulfillmentCount(): number {
+    return 1;
+  }
+
+  isNextExpectationMet(receivedEvent: Event): boolean {
+    const nextUnfulfilled = this.singleExpectations.find((expectation) =>
+      expectation.maybeFulfillExpectation(receivedEvent)
+    );
+
+    return nextUnfulfilled?.isFulfilled ?? false;
+  }
+
+  toString(): string {
+    return this.singleExpectations
+      .map((expectation) => expectation.toString())
+      .join(' - ');
+  }
+}
+
+export function EventSequence(
+  ...singleExpectationConvertibles: SingleEventExpectation[] | EventType[]
+): EventSequenceExpectation {
+  return new EventSequenceExpectation(singleExpectationConvertibles);
+}
+
+export function EventBag(
+  ...singleExpectationConvertibles: SingleEventExpectation[] | EventType[]
+): EventBagExpectation {
+  return new EventBagExpectation(singleExpectationConvertibles);
+}
+
+export function RepeatedEvent(
+  singleExpectationConvertible: SingleEventExpectation | EventType,
+  count: number
+): EventSequenceExpectation {
```

Comment:
```suggestion
): RepeatedEventExpectation {
```
Sorry, this one is on me. That was already the wrong type in my previous suggestion.